keps/sig-node/127-user-namespaces/README.md

```diff
@@ -363,88 +549,463 @@ Note this section is a WIP yet.
+protobuf][proto3-defaults]. We already tested this with real code.
+
+Old runtime and new kubelet: containers are created without userns. As the
+`user` field of the `NamespaceOption` message is not part of the runtime
+protofiles, that part is ignored by the runtime and pods are created using the
+host userns.
+
+
+[proto3-defaults]: https://developers.google.com/protocol-buffers/docs/proto3#default
+
 ## Production Readiness Review Questionnaire
 
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
 ### Feature Enablement and Rollback
 
-* **How can this feature be enabled / disabled in a live cluster?**
-  - [ ] Feature gate
-    - Feature gate name:
-    - Components depending on the feature gate:
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
 
-* **Does enabling the feature change any default behavior?**
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: UserNamespacesSupport
+  - Components depending on the feature gate: kubelet, kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+No.
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Pods that were created and opted-in to use user namespaces, will have to be
+recreated to run again without user namespaces.
+
+Other than that, it is safe to disable once it was enabled.
```

Comment:
>> Can the pods be safely deleted (without leaking anything) after the feature was disabled?
>>
>> Or should we ensure that those pods are deleted before disabling the feature gate?

> Good question.
>
> Yes, pods can be safely deleted even after the feature was disabled in the kubelet. We just store the mapping for this pod in /var/lib/kubelet/$POD/userns, the pod deletion takes care of deleting everything in that directory so we don't leak anything.
> 
> I've just tested it locally, to be sure I was not missing anything :)

Can you please:
- extend the text here that talks that pods using the feature can be deleted even after disbling the feature
- add an item to `Test Plan` section so that we will actualy add a test (integration?) that will be disabling the feature and then trying to delete the pods that was using it?

Reply:
Yes, thanks!

@wojtek-t I added that to the integration test. I'm not sure if it would be an integration or e2e test, though. But we will definitely test it. When we write the test, if it makes more sense to test it as an e2e test (unsure until I write the test :)), I'll open a PR to move this test to the e2e test section.

However, please note the comment that we have today on the e2e section. We basically need support from the runtime to run this test. We need to either see if we can use a patched version to run the tests (we will open PRs to containerd and CRIO as soon as this is accepted and we land the CRI changes, but it will probably take some time for these changes to be included in a containerd/CRIO release) or make alpha longer until we can run tests with released versions of the container runtimes that have this feature.

Reply:
Sure - I understand that. I just wanted to ensure that this won't get lost for future. I think it's important to ensure these tests to ensure that once we will be enabling it for beta we will know (and guarantee) the behavior to allow rolllback.